src/lib.rs

```diff
@@ -43,6 +43,14 @@ impl FromStr for AnimalName {
     }
 }
 
+#[cfg(feature = "crypto")]
+impl From<helium_crypto::PublicKey> for AnimalName {
+    fn from(pubkey: helium_crypto::PublicKey) -> Self {
+        // we unwrap since we know it can't fail
+        AnimalName::from_str(&pubkey.to_string()).unwrap()
```

Comment:
why not TryFrom instead?

Reply:
because it's infallible! My slightly repetitive code last time was:
```
 let digest = hex_digest(s);
 Self {
            adjective: words::ADJECTIVES[digest[0]],
            color: words::COLORS[digest[1]],
            animal: words::ANIMALS[digest[2]],
 }
 ```

Reply:
here i'll do something else. one sec

Reply:
ok - i pushed a refactoring that contains the infallible code. FromStr wraps it with Ok and From<PublicKey> can use it too without unwrapping.